### ISSUE TITLE
[Feature] Add support for "get"-style paths w/ constraint fields

### ISSUE THREAD

**Issue by arcanis** [yarnpkg/berry#69]
**Describe the user story**

I'm a project owner and I'd like to forbid my workspace projects from declaring postinstall scripts.

**Describe the solution you'd like**

I should be able to write the following rule:

```js
gen_workspace_field_requirement(_, 'scripts.install', null) :-
  workspace_field(WorkspaceField, 'scripts.install', _).
```

This isn't possible at the moment because it would look in a field named `"scripts.install"` rather than a field named `"install"` within an object named `"scripts"`.

**Describe the drawbacks of your solution**

It would require to either write a possibly complex logic, or add a new dependency to the constraints plugin.

**Describe alternatives you've considered**

I don't see much alternative apart from not supporting nested objects at all, which seems like a waste considering the potential use cases.

**Additional context**

The logic to access a specific field from an object is already implemented in [`_.get`](https://lodash.com/docs/#get). I haven't checked how many bytes it would add to the bundle, though.

**Comment by bgotink** [comment]
I was wondering about how to tackle this as well! There's another alternative I could think of, allowing for more complex checks: expose this via predicates, e.g.

```prolog
gen_workspace_field_requirement(WorkspaceCwd, 'scripts.install', null) :-
  workspace_field(WorkspaceCwd, 'scripts', ScriptsField),
  % the above line could be replaced with workspace_manifest(WorkspaceCwd, Manifest), value_has_property(Manifest, 'scripts', ScriptsField) to dedupe the ways properties have to be accessed
  value_is_object(ScriptsField),
  value_has_property(ScriptsField, 'install', _).
```

but maybe that ends up being too complex:

- we'd have to create a way to represent arbitrary JSON values in prolog
- a lot of extra prolog is necessary to do relatively simple stuff like the install scripts in the example

**Comment by arcanis** [comment]
Right, that would be possible as well 🤔

I think it would be best to keep the API as simple as possible, though - Prolog will already be a barrier to some users, we should make sure that part of the value they get in return will be a much lower amount of boilerplate than what they'd write in JS.

**Comment by bgotink** [comment]
`_.get` adds ± 5 kB minified or 22.8 kB unminified to the bundle size ([bundlephobia](https://bundlephobia.com/result?p=lodash.get@4.4.2)). Compared to `packages/berry-cli/bundles/berry.js`'s 4.7 MB that's not that much imo

**Comment by arcanis** [comment]
Yup, sounds reasonable 👍

**Comment by arcanis** [comment]
Implemented in #76 👍
